File: Leetcode/849-Maximize-Distance-to-Closest-Person/MaximizeDistancetoClosestPerson.js
/**
 * @param {number[]} seats
 * @return {number}
 */
var maxDistToClosest = function(seats) {
    let count = 0;
    let beforeStatus = -1
    let maxDis = 0;
    for (let i = 0; i < seats.length; i++) {
        if (seats[i] == 0) count++
            else {
                if (beforeStatus == -1 && count == 0) beforeStatus = seats[i];
                if (count % 2 == 1) {
                    if (beforeStatus == 1) maxDis = Math.max(Math.floor((count / 2) + 1), maxDis);
                    else {
                        maxDis = Math.max(count, maxDis);
                        beforeStatus = 1
                    }
                } else {
                    if (beforeStatus == 1) maxDis = Math.max(count / 2, maxDis);
                    else {
                        maxDis = Math.max(count, maxDis);
                        beforeStatus = 1
                    }
                }
                count = 0
            }
    };
    if (count > maxDis) maxDis = count
    return maxDis
};

File: Leetcode/849-Maximize-Distance-to-Closest-Person/MaximizeDistancetoClosestPerson.java
public class MaximizeDistancetoClosestPerson {
    public int maxDistToClosest(int[] seats) {
        int max = maxZeroDistance(seats);
        if (max % 2 == 1) {
            return max/2 + 1;
        }
        return max/2;
    }

    private int maxZeroDistance(int[] seats) {
        boolean isBeginning = true;
        int max = 0;
        int count = 0;
        for(int i=0; i<seats.length; i++) {
            if (seats[i] == 0) {
                count++;
            } else {
                if (isBeginning) {
                    max = Math.max(max, count*2);
                } else {
                    max = Math.max(max, count);
                }
                isBeginning = false;
                count=0;
            }
        }
        if (count != 0) {
            max = Math.max(max, count*2);
        }
        return max;
    }
}
